File: common/src/main/java/cloudcmd/common/engine/BlockCache.java
package cloudcmd.common.engine;

import cloudcmd.common.adapters.Adapter;

import java.util.List;
import java.util.Map;

public interface BlockCache
{
  Map<String, List<Adapter>> getHashProviders();

  Map<String, List<Adapter>> loadCache(int minTier, int maxTier) throws Exception;

  void refreshCache(int minTier, int maxTier) throws Exception;

  void init() throws Exception;

  void shutdown();

}

File: common/src/main/scala/cloudcmd/common/engine/BlockCache.scala
package cloudcmd.common.engine

import cloudcmd.common.adapters.Adapter

trait BlockCache {
  def getHashProviders(): Map[String, List[Adapter]]

  def loadCache(minTier: Int, maxTier: Int): Map[String, List[Adapter]]

  def refreshCache(minTier: Int, maxTier: Int)
}